sdk/main/src/Transaction.cc

```diff
@@ -844,6 +908,12 @@ void Transaction<SdkRequestType>::addTransaction(const proto::Transaction& trans
   signedTx.ParseFromArray(transaction.signedtransactionbytes().data(),
                           static_cast<int>(transaction.signedtransactionbytes().size()));
 
+  // will be true if the transaction came from bytes
+  if (signedTx.ByteSizeLong() == 0)
+  {
+    signedTx.set_bodybytes(transaction.SerializeAsString());
+  }
```

Comment:
This doesn't make sense to me. If the `signedTx` doesn't contain any body bytes (which I'm still confused on how that can happen), then you fill it with the bytes of the serialized `transaction`, which contains the serialized `signedTx`?